api/model/api.go

```diff
@@ -983,8 +985,61 @@ type RequestSigningMeta struct {
 	SignatureHeader string   `json:"signature_header"`
 }
 
-// GraphQLConfig is the root config object for a GraphQL API.
+type GraphQLFieldConfig struct {
+	TypeName              string   ` json:"type_name"`
+	FieldName             string   ` json:"field_name"`
+	DisableDefaultMapping bool     ` json:"disable_default_mapping"`
+	Path                  []string `json:"path"`
+}
+
+// +kubebuilder:validation:Enum=REST;GraphQL
+type GraphQLEngineDataSourceKind string
 
+type GraphQLEngineDataSource struct {
+	Kind       GraphQLEngineDataSourceKind `json:"kind"`
+	Name       string                      `json:"name"`
+	Internal   bool                        `json:"internal"`
+	RootFields []GraphQLTypeFields         `json:"root_fields"`
+	Config     json.RawMessage             `json:"config"`
```

Comment:
```suggestion
	Config     MapStringInterfaceType             `json:"config"`
```
If you will set the type of `Config` as `json.RawMessage`, in the CRD it will be set to string type.
Use `MapStringInterfaceType` instead as we want user to input json doc here.

Reply:
Updated 👍